### PR TITLE
:seedling: parameterizes the e2e target and pins opm images

### PR DESCRIPTION
**Pull request by perdasilva**
**Description of the change:**
The previous refactored parametrization of the e2e target was too restrictive. This PR re-introduces parameters to configure the e2e test command.

It has also come to our attention that opm:latest images are being used in e2e tests. This PR also pins the opm images to the relevant version of operator-registry

**Motivation for the change:**

**Architectural changes:**

**Testing remarks:**

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage
- [ ] Sufficient end-to-end test coverage
- [ ] Bug fixes are accompanied by regression test(s)
- [ ] e2e tests and flake fixes are accompanied evidence of flake testing, e.g. executing the test 100(0) times
- [ ] tech debt/todo is accompanied by issue link(s) in comments in the surrounding code
- [ ] Tests are comprehensible, e.g. Ginkgo DSL is being used appropriately
- [ ] Docs updated or added to `/doc`
- [ ] Commit messages sensible and descriptive
- [ ] Tests marked as `[FLAKE]` are truly flaky and have an issue
- [ ] Code is properly formatted

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -144,21 +157,22 @@ kind-create: kind-clean
...
+		--set catalog.opmImageArgs=--opmImage=$(OPERATOR_REGISTRY_IMAGE) \
 		--set package.image.ref=$(OLM_IMAGE) \
 		--set package.image.pullPolicy=IfNotPresent \
 		$(HELM_INSTALL_OPTS) \
 		--wait;
 
 .PHONY: e2e-build
 e2e-build: BUILD_TAGS="json1 e2e experimental_metrics"
-e2e-build: export GOOS=linux
-e2e-build: export GOARCH=amd64
```
Comment (tmshort) [inline Makefile:161]: Are you sure you don't need the `export`? We added that recently because it was causing problems with test builds on macs

File: Makefile [hunk 2]
```diff
@@ -144,21 +157,22 @@ kind-create: kind-clean
 OLM_IMAGE := quay.io/operator-framework/olm:local
 deploy:
 	$(KIND) load docker-image $(OLM_IMAGE) --name $(KIND_CLUSTER_NAME); \
-	$(HELM) install olm deploy/chart \
+	$(HELM) upgrade --install olm deploy/chart \
```
Comment (everettraven) [inline Makefile:160]: Could you elaborate on why this change was needed?

File: Makefile (hunk 1 above)
Comment (perdasilva) [reply Makefile:161]: I think we can drop the GOARCH as that could screw up M-class cpu mac people
I'll fix GOOS tho =D

File: Makefile (hunk 2 above)
Comment (perdasilva) [reply Makefile:160]: it was just convenient during development - if I just wanted to deploy and skip the whole kind create step

File: deploy/chart/values.yaml [hunk 3]
```diff
@@ -40,6 +40,7 @@ catalog:
   setWorkloadUserID: true
   replicaCount: 1
   commandArgs: --configmapServerImage=quay.io/operator-framework/configmap-operator-registry:latest
+  opmImageArgs: --opmImage=quay.io/operator-framework/opm:latest
```
Comment (everettraven) [inline deploy/chart/values.yaml:43]: I'm not very familiar with this code base, could you provide any resources about how this is injecting this image as the opm image to use?

Comment (perdasilva) [reply deploy/chart/values.yaml:43]: It's [this](https://github.com/operator-framework/operator-lifecycle-manager/pull/3275/commits/af279d96d68467c3808fc33e6aaaa4b5b7632ae0#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R257) bit that does the injection. This just sets a reasonable default for the value.

Comment (everettraven) [reply deploy/chart/values.yaml:43]: Apologies, I think I could have phrased this better - Where is this `--opmImage` flag defined and where is the provided value used within OLM?

By no means should this comment block this PR, but since I'm still relatively new to the codebase the additional context would be helpful for me :)

**Review by everettraven — APPROVED** [review]
Everything looks good to me from a code change perspective, but I'm not to familiar with this code base and how effective this change is. If we trust our e2e tests then LGTM :)

**Comment by tmshort** [comment]
unit tests are now failing :(

File: deploy/chart/values.yaml (hunk 3 above)
Comment (perdasilva) [reply deploy/chart/values.yaml:43]: All good, man. I'm learning myself 😂

This flag is defined by the catalog operator binary (in the cmd package)

**Comment by perdasilva** [comment]
> unit tests are now failing :( And the only change was the export‽‽

Flake (already investing XD - I'll get this repo stable if it kills me!)
